components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -413,18 +414,27 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
 
     private fun Uri.toFileUri(context: Context): Uri {
         val uri = this
-        val temporalFile = java.io.File(context.cacheDir, uri.getFileName(context))
+        val cacheUploadDirectory = java.io.File(context.cacheDir, "/uploads")
+
+        if (!cacheUploadDirectory.exists()) {
+            cacheUploadDirectory.mkdir()
+        }
+
+        val temporalFile = java.io.File(cacheUploadDirectory, uri.getFileName(context))
+        var inStream: InputStream? = null
+        var outStream: FileOutputStream? = null
         try {
-            val inStream = context.contentResolver.openInputStream(uri) as FileInputStream
-            val outStream = FileOutputStream(temporalFile)
+            inStream = context.contentResolver.openInputStream(uri) as FileInputStream
+            outStream = FileOutputStream(temporalFile)
             val inChannel = inStream.channel
             val outChannel = outStream.channel
             inChannel.transferTo(0, inChannel.size(), outChannel)
-            inStream.close()
-            outStream.close()
         } catch (e: IOException) {
             val logger = Logger("GeckoPromptDelegate")
             logger.warn("Could not convert uri to file uri", e)
+        } finally {
+            inStream?.close()
```

Comment:
Funny part about `Closeable` is that it [can throw exception](https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#close()) as well. :smile: 

What do you think about using [`use`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/use.html) instead (it's analogue to `try-with-resources` in Java)?

Reply:
👍